extra/corenrnmodl

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+# -----------------------------------------------------------------------------
+# corenrnmodl uses make to compile additonal mod files for coreneuron
+# Copyright 2019 Blue Brain Project
+# -----------------------------------------------------------------------------
+set -e
+APP_NAME=$(basename $0)
+_PARENT="$(dirname $BASH_SOURCE)/.."
+ROOTDIR=$(perl -e "use Cwd 'abs_path'; print abs_path('$_PARENT')")
+
+while getopts "n:v:o:i:l:h" OPT; do
+    case "$OPT" in
+    n)
+        params_MECH_NAME="$OPTARG";;
+    v)
+        params_MECH_VERSION="$OPTARG";;
+    o)
+        params_OUTPUT="$OPTARG";;
+    i)
+        params_INCFLAGS="$OPTARG";;
+    l)
+        params_LINKFLAGS="$OPTARG";;
+    h)
+        echo "$APP_NAME [options, ...] [mods_path]"
+        echo "Options:"
+        echo "  -n <name>       Optionally add mech name to library, as suffix"
+        echo "  -v <version>    Set libcorenrnmech version (default: 0.0)"
+        echo "  -o <output_dir> Set the output dir (default: <arch>-core)"
+        echo "  -i <incl_flags> Definitions passed to the compiler, typically '-I dir..'"
+        echo "  -l <link_flags> Definitions passed to the linker, typically '-Lx -lylib..'"
+        exit 0;;
+    ?)
+        exit 1;;
+    esac
+done
+
+shift $(($OPTIND - 1))
+
+if [ $# -gt 1 ]; then
+    echo "Error: $APP_NAME expects at most one mod dir. See syntax: '$APP_NAME -h' "
+    exit 1
+fi
+# If defined will be in $1
+params_MODS_PATH=$1
+
+
+make_params=()
+for param in MECH_NAME MECH_VERSION OUTPUT INCFLAGS LINKFLAGS MODS_PATH; do
+    var="params_${param}"
+    [ "${!var}" ] && make_params+=("$param=${!var}")
+done
+
+# Fail early if no mods
+ls "${params_MODS_PATH:-.}"/*.mod || { echo "No mods found"; exit 1; }
```

Comment:
doesn't neuron fail if there is no mod file provided? I don't remember...

Reply:
I will also work do this